Documentation/ceph-csi-drivers.md

```diff
@@ -175,3 +175,91 @@ CSI-Addons supports the following operations:
   - [Annotating PersistentVolumeClaims](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/reclaimspace.md#annotating-perstentvolumeclaims)
 - Network Fencing
   - [Creating a NetworkFence](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/networkfence.md)
+
+### Enable RBD Encryption Support
+
+Ceph-CSI supports encrypting RBD volumes with LUKS encryption. More details can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/docs/deploy-rbd.md#encryption-for-rbd-volumes)
+with full list of supported encryption configurations. A sample configmap can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/examples/kms/vault/kms-config.yaml).
+
+Following steps demonstrate how to enable support for encryption:
+
+* Create `rook-ceph-csi-kms-config` configmap with required encryption configuration in
+the same namespace where Rook operator is deploy. An example is shown below:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-kms-config
+  namespace: rook-ceph
+data:
+  config.json: |-
+    {
+      "user-secret-metadata": {
+        "encryptionKMSType": "metadata",
+        "secretName": "storage-encryption-secret"
+      }
+    }
+```
```

Comment:
Why is the name of this configmap hard-coded but the field inside (`user-secret-metadata`) configurable?

Reply:
Is it so different storage classes can each use their own entries? But a single configmap only needs to be loaded by the csi driver.

Reply:
Could this configmap be avoided if the encryptionKMSType and secretName properties were just in the storageclass?

Reply:
> Could this configmap be avoided if the encryptionKMSType and secretName properties were just in the storageclass?

no, storageclass parameters are not available during various ops, 
Ceph-csi would use encryptionKMSID( this is stored in omap) to fetch these options from the configmap.


> Why is the name of this configmap hard-coded but the field inside (`user-secret-metadata`) configurable?

This is because of the following. 
> Is it so different storage classes can each use their own entries? But a single configmap only needs to be loaded by the csi driver.

Yes

Reply:
> > Could this configmap be avoided if the encryptionKMSType and secretName properties were just in the storageclass?
> 
> no, storageclass parameters are not available during various ops, Ceph-csi would use encryptionKMSID( this is stored in omap) to fetch these options from the configmap.

I'm not understanding yet. If the encryptionDMSID is needed from the storage class to look up values in the configmap, seems like the storage class could just specify those properties directly. But let's discuss in the weekly call to clarify, thanks.

Reply:
Thanks for the offline explanation of the configmap. Let's go ahead and continue the review on other comments and get the merge conflicts resolved, thanks!